packages/keiko-ui/src/app/components/desktop/widgets/coding-workbench/CodingWorkbenchWindow.tsx

```
@@ -883,11 +984,54 @@ function ChangesetReviewPanel({
+  loading: "codingWorkbench.approval.changes.loading",
+  unavailable: "codingWorkbench.approval.changes.unavailable",
+  retry: "codingWorkbench.approval.changes.retry",
+} as const;
+
+const RESEARCH_DESTINATION_MESSAGE_KEYS = {
+  loading: "codingWorkbench.approval.research.loading",
+  unavailable: "codingWorkbench.approval.research.unavailable",
+  retry: "codingWorkbench.approval.research.retry",
+} as const;
+
 function ApprovalChangedFiles({
```

ESLint

[error] Function 'ApprovalChangedFiles' has too many lines (57). Maximum allowed is 50.

(max-lines-per-function)